### PR TITLE
Redux/saving

### PR DESCRIPTION
**Pull request by evanharmon**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,8 @@
     "redux-thunk": "^2.2.0"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "json-server --static ./build db.json",
+    "dev": "react-scripts start",
```
Comment (evanharmon) [inline package.json]: @laurelherting `yarn start` is now reserved for production use. You'll need to do `yarn dev` in one terminal and `yarn run dev-server` in another terminal.

File: src/components/NoteList.js [hunk 2]
```diff
@@ -1,23 +1,14 @@
...
-  title: PropTypes.string,
-};
+import {
+  fetchNotes,
+  saveNote,
+  updateCurrent,
+  toggleNote,
+  deleteNote,
+} from '../reducers/note';
+import NoteItem from './Note';
```
Comment (evanharmon) [inline src/components/NoteList.js]: @laurelherting refactored NoteItem component to separate file

File: src/reducers/note.js [hunk 3]
```diff
@@ -29,6 +40,22 @@ export const saveNote = title => async dispatch => {
   dispatch(addNote(res));
 };
 
+export const toggleNote = id => async (dispatch, getState) => {
```
Comment (evanharmon) [inline src/reducers/note.js]: @laurelherting this isn't really fully fleshed out yet. At the moment it saves

File: src/components/Note.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const NoteItem = ({ id, title, toggleNote, deleteNote }) => (
+  <li>
+    <span className="delete-item">
+      <button onClick={() => deleteNote(id)}>X</button>
```
Comment (laurelherting) [inline src/components/Note.js]: Is the deleteNote(id) called down from const?

File: src/components/NoteList.js [hunk 5]
```diff
@@ -52,7 +43,14 @@ class NoteList extends Component {
           />
         </form>
         <ul>
-          {this.props.notes.map(note => <NoteItem key={note.id} {...note} />)}
+          {this.props.notes.map(note => (
+            <NoteItem
+              key={note.id}
+              toggleNote={this.props.toggleNote}
+              deleteNote={this.props.deleteNote}
```
Comment (evanharmon) [inline src/components/NoteList.js]: @laurelherting This is where i'm passing `deleteNote` down to the individual NoteItems in the file `Note.js`

**Pull request merged**